src/write.jl

```diff
@@ -442,40 +442,62 @@ function _write_constraint(
     return
 end
 
-function _write_nlp_constraint(io::IO, variables, expr::Expr)
+function _write_constraint(
+    io::IO,
+    variables,
+    f::MOI.ScalarNonlinearFunction{T},
+    s::MOI.EqualTo{T},
```

Comment:
What about sets other than `EqualTo`? By default we'd support all the scalar sets: `LessThan`, `GreaterThan`, `ZeroOne`, etc.

Reply:
yeah, though the old code only allowed RHS of true/false, whereas really it should be any `T`. i'll try to loosen that restriction and then allow the other types of scalar sets. it seemed weird to me to allow `> true` for example.

Reply:
I think it should be sufficient to write the function and set part separately. See the `ScalarAffineFunction` method above on line 415.

Reply:
fixed